src/cpu/x64/ukernel/brgemm.cpp

```diff
@@ -186,8 +201,17 @@ status_t brgemm_t::generate() {
 
 status_t brgemm_t::execute(const void *A_ptr, const void *B_ptr,
         const dim_t *A_B_offsets, void *C_ptr, void *scratchpad_ptr) const {
+    // Assert scratchpad is cache-line aligned
+    assert(reinterpret_cast<uintptr_t>(scratchpad_ptr) % k_cache_line == 0);
```

Comment:
`assert` might spoil user's experience in case they don't have it aligned. Probably a verbose output with warning like [this one](https://github.com/uxlfoundation/oneDNN/blob/main/src/cpu/jit_utils/jit_utils.cpp#L100-L101) would be a better option.